### PR TITLE
Bash auto completion

### PR DESCRIPTION
**Pull request by alexcb**
To enable auto completion, a user must enter:

  complete -o filenames -C "/path/to/earth" earth

### CODE REVIEW

File: autocomplete/complete_test.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	matches, err := GetPotentials("earth pru", 9, nil, []string{"prune", "foo"})
+	NoError(t, err, "GetPotentials failed")
+	Equal(t, []string{"prune "}, matches)
+}
+
+func TestPathCompletion(t *testing.T) {
+	matches, err := GetPotentials("earth .", 7, nil, nil)
+	NoError(t, err, "GetPotentials failed")
+	Equal(t, []string{"./", "../"}, matches)
+}
```
Comment (vladaionescu) [inline autocomplete/complete_test.go:25]: Very nice

File: cmd/earth/main.go [hunk 2]
```diff
@@ -79,7 +81,66 @@ var (
...
+	}
+
+	commands := []string{}
+	for _, cmd := range app.cliApp.Commands {
+		commands = append(commands, cmd.Name)
+	}
+
+	potentials, err := autocomplete.GetPotentials(compLine, int(compPointInt), flags, commands)
+	if err != nil {
+		//panic(err) // can't display error or it will show up under tab-completion
```
Comment (vladaionescu) [inline cmd/earth/main.go:116]: Would stderr work for errors?

I assume that the answer is "no" given that you probably tried `panic`.

File: cmd/earth/main.go [hunk 3]
```diff
@@ -79,7 +81,66 @@ var (
 	GitSha string
 )
 
+// to enable autocomplete, enter
+// complete -o nospace -C "/path/to/earth" earth
+func autoComplete(app *earthApp) {
```
Comment (vladaionescu) [inline cmd/earth/main.go:86]: How about

```suggestion
func (app *earthApp) autoComplete() {
```

File: cmd/earth/main.go (hunk 2 above)
Comment (alexcb) [reply cmd/earth/main.go:116]: unfortunately not; if the panic is there it'll suggestion the output as options.

**Pull request merged**